### ISSUE TITLE
Crash in Converter::InterpolateKeys converting attached FBX file

### ISSUE THREAD

**Issue by griffin2000** [assimp/assimp#2088]
The attached FBX will crash Assimp at 2879e0d6ef21a34180da13dd9f0a66b16d54b095.

Simply calling assimp_cmd dump with file as input shows the crash.

The error caused by the a zero in the last field passed to Converter::InterpolateKeys inputs so line 2849 will produce an index of -1.  

> const size_t id1 = next_pos[ i ] == ksize ? ksize - 1 : next_pos[ i ];

I've also attached a minidump from Visual Studio 2015, just before the crash occurs.

[assimp.zip](https://github.com/assimp/assimp/files/2272089/assimp.zip)

[GirlRun.zip](https://github.com/assimp/assimp/files/2272072/GirlRun.zip)

**Comment by griffin2000** [comment]
Wrapping that whole section in if(ksize>0) gets rid of the crash. Don't know how valid the results are though (will mean nothing is put in the result array): 

>             if (ksize > 0) {
>                 const size_t id0 = next_pos[i]>0 ? next_pos[i] - 1 : 0;
>                 const size_t id1 = next_pos[i] == ksize ? ksize - 1 : next_pos[i];
> 
>                 // use lerp for interpolation
>                 const KeyValueList::value_type valueA = std::get<1>(kfl)->at(id0);
>                 const KeyValueList::value_type valueB = std::get<1>(kfl)->at(id1);
> 
>                 const KeyTimeList::value_type timeA = std::get<0>(kfl)->at(id0);
>                 const KeyTimeList::value_type timeB = std::get<0>(kfl)->at(id1);
> 
>                 const ai_real factor = timeB == timeA ? ai_real(0.) : static_cast<ai_real>((time - timeA)) / (timeB - timeA);
>                 const ai_real interpValue = static_cast<ai_real>(valueA + (valueB - valueA) * factor);
> 
>                 result[std::get<2>(kfl)] = interpValue;
>             }

**Comment by kimkulling** [comment]
Good catch. I just checked the model in blender and it seems, that it only contains the animation data. I do not see any geometry ( on a first and fast look).
I will prepare a pull request for the workaround. Thanks a lot for figuring this out.

**Comment by griffin2000** [comment]
Yes, there is no geometry associated with the file, just animation data. Though that alone is not enough to cause the crash.
